src/test/java/org/sagebionetworks/bridge/spring/controllers/MasterSchedulerControllerTest.java

```diff
@@ -0,0 +1,242 @@
+        doReturn(session).when(controller).getAuthenticatedSession(ADMIN);
+        
+        mockRequestBody(mockRequest, mockConfig);
+        
+        // Mock service.
+        when(mockSchedulerService.createSchedulerConfig(any())).thenReturn(mockConfig);
+        
+        // Execute and validate.
+        MasterSchedulerConfig result = controller.createSchedulerConfig();
+        
+        assertEquals(result.getScheduleId(), SCHEDULE_ID);
+
+        // Verify service.
+        verify(mockSchedulerService).createSchedulerConfig(configCaptor.capture());
+        
+        MasterSchedulerConfig config = configCaptor.getValue();
+        assertEquals(config.getScheduleId(), SCHEDULE_ID);
+        assertEquals(config.getCronSchedule(), CRON_SCHEDULE);
+        assertTrue(config.getRequestTemplate().get("a").booleanValue());
+        assertEquals(config.getRequestTemplate().get("b").textValue(), "string");
+        assertEquals(config.getSqsQueueUrl(), SQS_QUEUE_URL);
+        assertEquals(config.getVersion(), new Long(VERSION));
+    }
+    
+    @Test(expectedExceptions = UnauthorizedException.class)
+    public void createSchedulerConfigAdminOnly() throws Exception {
+        session.setParticipant(new StudyParticipant.Builder().withRoles(ImmutableSet.of(DEVELOPER)).build());
+        
+        controller.createSchedulerConfig();
+        
+        verify(mockSchedulerService, never()).createSchedulerConfig(any());
+    }
+    
+    @Test
+    public void getSchedulerConfig() throws Exception {
+        session.setParticipant(new StudyParticipant.Builder().withRoles(ImmutableSet.of(ADMIN)).build());
+        doReturn(session).when(controller).getAuthenticatedSession(ADMIN);
+        
+        // Mock service.
+        when(mockSchedulerService.getSchedulerConfig(SCHEDULE_ID)).thenReturn(mockConfig);
+        
+        // Execute and validate.
+        MasterSchedulerConfig result = controller.getSchedulerConfig(SCHEDULE_ID);
+        assertEquals(result.getScheduleId(), SCHEDULE_ID);
+        assertSame(mockConfig, result);
+        verify(controller).getSchedulerConfig(SCHEDULE_ID);
+    }
+    
+    @Test
+    public void updateSchedulerConfig() throws Exception {
+        session.setParticipant(new StudyParticipant.Builder().withRoles(ImmutableSet.of(ADMIN)).build());
+        doReturn(session).when(controller).getAuthenticatedSession(ADMIN);
+        
+        MasterSchedulerConfig updatedConfig = MasterSchedulerConfig.create();
+        updatedConfig.setScheduleId("test-schedule-id");
+        updatedConfig.setCronSchedule(UPDATED_CRON_SCHEDULE);
+        updatedConfig.setRequestTemplate(mockConfig.getRequestTemplate());
+        updatedConfig.setSqsQueueUrl(UPDATED_SQS_QUEUE_URL);
+        updatedConfig.setVersion(2L);
+        mockRequestBody(mockRequest, updatedConfig);
+        
+        // Mock service.
+        when(mockSchedulerService.getSchedulerConfig(SCHEDULE_ID)).thenReturn(mockConfig);
```

Comment:
You don't need to mock schedulerService.get(), because since it's a mock, it will never be called. Also, we don't want implementation details from the service to leak into our tests for the controller.